core/capstanignore.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package core
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Capstanignore interface {
+	LoadFile(path string) error
+	AddPattern(pattern string) error
+	PrintPatterns()
+	IsIgnored(path string) bool
+}
+
+var CAPSTANIGNORE_ALWAYS []string = []string{"/meta/*", "/mpm-pkg/*", "/.git/*"}
+
+// CapstanignoreInit creates a new capstanignore struct that is
```

Comment:
I see now why lowecase is used. I advise not to mention a struct that is not exposed as all it matters is that this initialiser will return a struct compliant with the `Capstanignore` interface.